crates/editor_ui/src/inspector/events.rs

```diff
@@ -1,40 +1,41 @@
 use bevy::prelude::*;
 
-use bevy_egui::*;
+use bevy_egui::{egui::Color32, *};
 
 use crate::prelude::*;
 
 #[derive(Resource, Default)]
-pub struct EventTab;
+pub struct EventDebuggerTab;
 
-impl EditorTab for EventTab {
+impl EditorTab for EventDebuggerTab {
     fn ui(&mut self, ui: &mut egui::Ui, _: &mut Commands, world: &mut World) {
         inspect(ui, world);
     }
 
     fn title(&self) -> egui::WidgetText {
-        "Event".into()
+        "Event Debugger".into()
     }
 }
 
 pub fn inspect(ui: &mut egui::Ui, world: &mut World) {
-    let mut events = world
-        .resource::<EditorRegistry>()
-        .send_events
-        .iter()
-        .map(|(type_id, event)| (*type_id, event.clone()))
-        .collect::<Vec<_>>();
-
-    events.sort_by(|(_, a), (_, b)| a.name().cmp(b.name()));
-
-    egui::Grid::new("Events ID".to_string()).show(ui, move |ui| {
-        for (type_id, event) in events.into_iter() {
-            ui.push_id(format!("{:?}-{}", &type_id, event.name()), |ui| {
-                if ui.button(event.name()).clicked() {
-                    event.send(world);
-                }
-            });
-            ui.end_row();
-        }
-    });
+    let events = &world.resource::<EditorRegistry>().send_events.clone();
+
+    if events.is_empty() {
```

Comment:
I think this can be:

```rs
 let events = &world.resource::<EditorRegistry>().send_events.iter();

 if events.count() == 0 { // Clippy might suggest something better
```

Reply:
I tried, but to no avail. The issue is `world` is needed for both accessing the `EditorRegistry` resource and sending the event. Not sure how to please the borrow checker without cloning.